editor/src/components/canvas/design-panel-root.tsx

```diff
@@ -183,26 +191,15 @@ export const DesignPanelRoot = betterReactMemo('DesignPanelRoot', (props: Design
     [interfaceDesigner, navigatorVisible, props.isUiJsFileOpen],
   )
 
-  const getNavigatorLeft = React.useMemo((): number | undefined => {
-    const codeEditorCurrentWidth =
-      codeEditorResizingWidth != null ? codeEditorResizingWidth : interfaceDesigner.codePaneWidth
-    let position = navigatorVisible ? codeEditorCurrentWidth : undefined
+  const [navigatorWidth, setNavigatorWidth] = usePubSubAtom(NavigatorWidthAtom)
 
-    if (!interfaceDesigner.codePaneVisible) {
-      if (leftMenuExpanded) {
-        position = LeftPaneDefaultWidth
-      } else {
-        position = undefined
-      }
-    }
-    return position
-  }, [
-    codeEditorResizingWidth,
-    interfaceDesigner.codePaneVisible,
-    interfaceDesigner.codePaneWidth,
-    leftMenuExpanded,
-    navigatorVisible,
-  ])
+  const onNavigatorResizeStop = React.useCallback<ResizeCallback>(
+    (_event, _direction, _ref, delta) => {
+      setNavigatorWidth((currentWidth) => currentWidth + delta.width)
+      dispatch([CanvasActions.scrollCanvas(canvasPoint({ x: -delta.width, y: 0 }))])
```

Comment:
This line feels like it would be a pretty jarring experience tbh :grimacing:

Reply:
it's quite nice actually